reference/basic-architecture.md

```diff
@@ -9,61 +9,66 @@ More detailed guides to Mayastor's components, their design and internal structu
 | Name | Resource Type | Function | Frequency in Cluster |
 | :--- | :--- | :--- | :--- |
 | **Control Plane** |
-| core-agent | Pod | Principle control plane actor | Single |
-| csi-controller | Pod | Hosts Mayastor's CSI controller implementation and CSI provisioner side car| Single |
-| rest | Pod | Hosts the public API REST server | Single |
-| *rest*  | Service | Exposes the REST API server via NodePort |
-| msp-operator | Pod | Hosts Mayastor's pool operator | Single |
-| mayastor-csi| Pod | Hosts CSI Driver node plugin containers | All worker nodes |
-| nats | Pod | Hosts NATS Server container | Single |
-| *nats* | Service | Exposes NATS message bus end point | Single |
-| *nats-config* | ConfigMap | NATS cluster configuration data |
-| etcd | Pod | Hosts etcd Server container | Single |
-| *mayastor-etcd* | Service | Exposes NATS message bus end point | Single |
-| *mayastor-etcd-headless* | Service | Exposes NATS message bus end point | Single |
-| **Data Plane**|
-| mayastor| Pod | Hosts Mayastor I/O engine| User-selected nodes |
-| **k8s Resource Types**  |
-| mayastorpools | CRD | Declares a Mayastor pool's desired state and reflects its current state | User-defined, one or many |
+| agent-core | Deployment | Principle control plane actor | Single |
+| csi-controller | Deployment | Hosts Mayastor's CSI controller implementation and CSI provisioner side car| Single |
+| api-rest | Pod | Hosts the public API REST server | Single |
+| *api-rest*  | Service | Exposes the REST API server via NodePort |
+| operator-diskpool | Deployment | Hosts DiskPool operator | Single |
+| csi-node| DaemonSet | Hosts CSI Driver node plugin containers | All worker nodes |
+| etcd | StatefulSet | Hosts etcd Server container | Configurable(<i>Recommended: Three replicas)</i> |
+| etcd | Service | Exposes etcd DB endpoint | Single |
+| etcd-headless | Service | Exposes etcd DB endpoint | Single |
+| io-engine| DaemonSet | Hosts Mayastor I/O engine| User-selected nodes |
+| DiskPool | CRD | Declares a DiskPool's desired state and reflects its current state | User-defined, one or many |
+| **Additional services**  |
```

Comment:
Additional components instead of services, to distinguish from the K8s service

Reply:
Updated it.